zone/spells.cpp

```diff
@@ -440,6 +440,32 @@ bool Mob::DoCastSpell(uint16 spell_id, uint16 target_id, CastingSlot slot,
 	// ok now we know the target
 	casting_spell_targetid = target_id;
 
+	if (IsInvisSpell(spell_id) && RuleB(Spells, InvisRequiresGroup)) {
+
+		Mob* target = GetTarget();
+
+		if (target && target->IsClient()) {
+			Client* spelltarget = entity_list.GetClientByID(target_id);
```

Comment:
Should call `target->CastToClient()` instead. Mobs can also be in groups (bots, mercs) this should be accounted for as well.

Reply:
So instead of Client* spelltarget I need to use a mob method and account for IsBot & IsClient or something of that nature